pallene/to_ir.lua

```diff
@@ -775,7 +780,9 @@ function ToIR:exp_to_assignment(cmds, dst, exp)
             elseif src_typ._tag == "types.T.Any" then
                 table.insert(cmds, ir.Cmd.FromDyn(loc, dst_typ, dst, v))
             else
-                error("impossible")
+                typedecl.tag_error(tag,
+                    string.format("error casting from type '%s' to '%s'",
+                        types.tostring(src_typ), types.tostring(dst_typ)))
```

Comment:
I think the `tag_error(tag` is not appropriate here, because the tags that actually matter for the error are `dst_typ._tag` and `src_typ.tag` and not the `exp._tag`.

I like the new error message but I think we could just use a plain `error(string.format(` instead.